edb/server/bootstrap.py

```diff
@@ -250,13 +243,77 @@ async def _is_pristine_cluster(ctx: BootstrapContext) -> bool:
         return False
 
 
+async def _is_pristine_cluster(ctx: BootstrapContext) -> bool:
```

Comment:
There's a lot of branching here, can we simplify this method to look for single-db metadata specifically, and, if not found, fallback to regular tenant detection?

Reply:
Yeah, with the `--with-backend-capability`/`--without-backend-capability` options to be added, I think we should raise an error here in `ensure_bootstrapped()` if:

1. The backend capability indicates e.g. the lack of `CREATE DATABASE` permission, and
2. The backend is already bootstraped differently, e.g. not in single-db mode

And the `_is_pristine_cluster()` only needs to check for single-db metadata like you said. I'll update the PR.

Reply:
UPDATE: Oh no, the regular detection couldn't find out if the backend is already bootstrapped in a single-role mode, because it's only checking the roles of `[<tenant_hash>_]edgedb_supergroup`. In this case, continuting to bootstrap as a pristine cluster will fail.

Reply:
Pushed a fix to 1) raise errors and 2) run regular tenant detection first, then check single modes in an easier way.